pymc_marketing/mmm/delayed_saturated_mmm.py

```diff
@@ -515,33 +513,24 @@ def channel_contributions_forward_pass(
         array-like
             Transformed channel data.
         """
-        alpha_posterior = self.fit_result["alpha"].to_numpy()
-
-        lam_posterior = self.fit_result["lam"].to_numpy()
-        lam_posterior_expanded = np.expand_dims(a=lam_posterior, axis=2)
-
-        beta_channel_posterior = self.fit_result["beta_channel"].to_numpy()
-        beta_channel_posterior_expanded = np.expand_dims(
-            a=beta_channel_posterior, axis=2
-        )
-
-        geometric_adstock_posterior = geometric_adstock(
-            x=channel_data,
-            alpha=alpha_posterior,
-            l_max=self.adstock_max_lag,
-            normalize=True,
-            axis=0,
-        )
+        coords = {
+            **self.model_coords,
+            **self.coords_mutable,
+        }
+        with pm.Model(coords=coords):
+            pm.Deterministic(
+                "channel_contributions",
+                self.forward_pass(x=channel_data),
+                dims=("date", "channel"),
+            )
 
-        logistic_saturation_posterior = logistic_saturation(
-            x=geometric_adstock_posterior,
-            lam=lam_posterior_expanded,
-        )
+            idata = pm.sample_posterior_predictive(
+                self.fit_result,
+                var_names=["channel_contributions"],
+                progressbar=False,
+            )
 
-        channel_contribution_forward_pass = (
-            beta_channel_posterior_expanded * logistic_saturation_posterior
-        )
-        return channel_contribution_forward_pass.eval()
+        return idata.posterior_predictive.channel_contributions.to_numpy()
```

Comment:
This seems to still make the progress bars and since it is done in the loop for many values, this seems annoying. 

Another implementation we could use could be making use of xarray broadcasting

Do you have any opinions here @juanitorduz

Reply:
Indeed! This step has to be done carefully as we are passing different inputs to the pytensor variables transformations and then evaluating (`.eval`) at the very end to get the xarray.